### PR TITLE
openwisp-monitoring: fix Makefile for 0.2.0

### PR DESCRIPTION
**Pull request by pandafy**
Maintainer: @nemesifier 
Compile tested: ramips
Run tested: ramips

### CODE REVIEW

**Review by nemesifier — APPROVED** [review]
LGTM :heavy_check_mark:

**Review by nemesifier — COMMENTED** [review]
@feckert we fixed some issues and updated some outdated code here.

**Comment by BKPepe** [comment]
Sure, maintainer could merge this, but commit subject seems vague to me and there is missing commit description.

File: admin/openwisp-monitoring/Makefile [hunk 1]
```diff
@@ -26,18 +26,25 @@ define Package/openwisp-monitoring
   SECTION:=admin
   SUBMENU:=openwisp
   DEPENDS:=+netjson-monitoring +openwisp-config
+  PKGARCH:=all
   URL:=http://openwisp.org
 endef
 
 define Package/netjson-monitoring
-  TITLE:=NetJson Monitoring
+  TITLE:=NetJSON Monitoring
```
[the remark this answers is not in the document]
Comment (pandafy) [reply admin/openwisp-monitoring/Makefile:33]: The commit message and PR title got mixed up with #25186. I apologise for the confusion. I have updated the commit message and added description explaining the need for this patch.

File: admin/openwisp-monitoring/Config.in [hunk 2]
```diff
@@ -0,0 +1,9 @@
+menu "netjson-monitoring Configuration"
+
+config NETJSON_MONITORING_IWINFO
+	bool "Enable rpcd-mod-iwinfo"
+	default y
+	help
+	  Whether to include the rpcd-mod-iwinfo dependency (enabled by default).
```
[the remark this answers is not in the document]
Comment (nemesifier) [reply admin/openwisp-monitoring/Config.in:7]: @BKPepe this is necessary for devices which do not have a WiFi card and have limited space, as it ensures the monitoring tools function without errors and optimizes space usage.

[the remark this answers is not in the document]
Comment (BKPepe) [reply admin/openwisp-monitoring/Config.in:7]: As it is enabled by default and that is not changed, I am curious about the space usage difference, though.

**Review by nemesifier — COMMENTED** [review]
it's ready for review again @BKPepe

File: admin/openwisp-monitoring/Config.in (hunk 2 above)
[the remark this answers is not in the document]
Comment (nemesifier) [reply admin/openwisp-monitoring/Config.in:7]: @BKPepe is not only a space usage problem, it's something we need for some use cases where the WiFi card is not there and having these modules enabled is causing issues in the output which we can't solve on our own so we just disable it. Is this a blocker for merging?

**Comment by feckert** [comment]
The two commits are not quite right.
They are not separate! The new option `NETJSON_MONITORING_IWINFO` should be added in a seperate commit but the variable is already used in the first [commit](https://github.com/openwrt/packages/pull/25187/commits/a8406803843c0a1b6182adef7deedccc496a0c86#diff-397895975bf552fbe2b81022675294f0469acb443e6560b37464ee6946651f7dR98)!

You should also not use `sed` when evaluating the variable `CONFIG_NETJSON_MONITORING_IWINFO`.
It can also be easier. Please use the following syntax https://github.com/openwrt/packages/blob/master/net/gensio/Makefile#L198

**Comment by BKPepe** [comment]
Good catch here, @feckert! :)

**Comment by pandafy** [comment]
@BKPepe @feckert I have made the requested changes on this PR.

**Comment by feckert** [comment]
I would also suggest to bump the PKG_RELEASE by one. Everything else looks good to me and is ready to merge.

File: admin/openwisp-monitoring/Makefile [hunk 3]
```diff
@@ -6,7 +6,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=openwisp-monitoring
 PKG_VERSION:=0.2.0
-PKG_RELEASE:=1
+PKG_RELEASE:=2
```
Comment (pandafy) [inline admin/openwisp-monitoring/Makefile:9]: @feckert I have bumped the PKG_RELEASE. It is part of the second commit. Is that okay?

Comment (feckert) [reply admin/openwisp-monitoring/Makefile:9]: Fine with me

**Comment by feckert** [comment]
Thanks merged!

**Pull request merged**